internal/handlers/zip.go

```diff
@@ -1,37 +1,38 @@
 package handlers
 
 import (
-	"errors"
-	"github.com/gorilla/mux"
 	"log"
 	"net/http"
-	"opg-file-service/dynamo"
-	"opg-file-service/internal"
-	"opg-file-service/middleware"
-	"opg-file-service/session"
-	"opg-file-service/zipper"
 	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/ministryofjustice/opg-file-service/internal/storage"
 )
 
+type Repository interface {
```

Comment:
just a question really - is it a best practice in go to have multiple interfaces defined in the same file? trying to understand coming from a c# background, and the rationale behind this having seen this in other go code too...

Reply:
Yeah, generally the idea is to put things where they are used. This interface is only used in `ZipHandler` so it makes sense to keep it close by. Take a look at [this example](https://github.com/Evertras/go-interface-examples/blob/master/local-interfaces/handlers/user.go), and [the README](https://github.com/Evertras/go-interface-examples/tree/master/local-interfaces) for an explanation.

Reply:
> Imagine if instead there was some IDb interface from a traitorous C# coder that contained 50 different function signatures, but all leaderboard ever needed was that one.

I laughed

Reply:
> > Imagine if instead there was some IDb interface from a traitorous C# coder that contained 50 different function signatures, but all leaderboard ever needed was that one.
> 
> I laughed

remember I _was_ that traitorous c# coder 🤣